File: app/src/main/java/com/mindorks/bootcamp/learndagger/ui/HomeFragment.java
package com.mindorks.bootcamp.learndagger.ui;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.mindorks.bootcamp.learndagger.MyApplication;
import com.mindorks.bootcamp.learndagger.R;
import com.mindorks.bootcamp.learndagger.di.component.DaggerFragmentComponent;
import com.mindorks.bootcamp.learndagger.di.module.FragmentModule;

import javax.inject.Inject;

public class HomeFragment extends Fragment {
    public static final String TAG = "HomeFragment";

    @Inject
    public HomeViewModel homeViewModel;

    public static HomeFragment newInstance() {
        return new HomeFragment();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getDependencies();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_home, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        Log.d(TAG, homeViewModel.getSomeData());
    }

    private void getDependencies() {
        Activity activity = getActivity();
        if (activity != null) {
            DaggerFragmentComponent
                    .builder()
                    .fragmentModule(new FragmentModule())
                    .applicationComponent(((MyApplication) activity.getApplication()).applicationComponent)
                    .build()
                    .inject(this);
        }
    }
}

File: app/src/main/java/com/mindorks/bootcamp/learndagger/ui/HomeFragment.kt
package com.mindorks.bootcamp.learndagger.ui

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.mindorks.bootcamp.learndagger.MyApplication
import com.mindorks.bootcamp.learndagger.R
import com.mindorks.bootcamp.learndagger.di.component.DaggerFragmentComponent
import com.mindorks.bootcamp.learndagger.di.module.FragmentModule
import javax.inject.Inject

class HomeFragment : Fragment() {
    companion object {
        val TAG = HomeFragment::class.java.canonicalName
        fun newInstance() = HomeFragment()
    }

    @Inject
    lateinit var homeViewModel: HomeViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getDependencies()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_home, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        Log.d(TAG, homeViewModel.getSomeData())
    }

    private fun getDependencies() {
        activity?.let {
            DaggerFragmentComponent
                    .builder()
                    .fragmentModule(FragmentModule())
                    .applicationComponent((it.application as MyApplication).applicationComponent)
                    .build()
                    .inject(this)
        }
    }
}